### DISCUSSION TITLE
How to load a single mf6 package not associated with a model

### DISCUSSION THREAD

**Discussion by RyanConway91** [modflowpy/flopy#1913] [Q&A]
I have MF6 simulation with several models using the gwf-gwf exchange. I just want to load a single package from one model. Loading the whole simulation is very slow. Can I do this?

I am thinking a potential process might be to make a empty "dummy" model and just load the package to that? However I cannot find a method to load an existing package file.

**Comment by javgs-bd** [comment]
I have wondered about this too in the past!

**Comment by wpbonelli** [comment]
You can use `load_only` to specify a subset of packages by type or name, for example

```python
sim = MFSimulation.load(
    "my_sim", sim_ws="/path/to/my/ws", load_only=["ic", "maw"]
)
```

Then retrieve the package from its parent model as usual. There is currently no way to specify a subset of models, only packages, to my knowledge.

For usage examples see https://github.com/modflowpy/flopy/blob/ed3a0cd68ad1acb68fba750f415fa095a19b741a/autotest/regression/test_mf6.py#L4391

I don't see this documented anywhere besides the [release notes](https://github.com/modflowpy/flopy/blob/develop/docs/version_changes.md#version-321), it would be good to though as it seems a common use case

**Reply by javgs-bd** [reply]
Thanks @w-bonelli I wonder if this is because to load a package it is needed to know the whole structure coming from the nam file and not only from the package file. Therefore you always have to create a sim/model for loading just a package. 

I wonder if there is no way to just use the existing structure of the nam in the preloaded model and ask flopy to just load the package from a new package file, and register it with the model. Also, I guess for non MF6 you just have to create another model object.

**Comment by SercanC** [comment]
I apologize for hijacking this thread, but I have a related question. When loading one package, is there a way to not load the time series file? Specifically, I want to load the RIV package to update the conductance values, and the stage time series are stored in a separate time series file. The RIV table is loaded quickly but it takes forever to load the time series in my case.

**Reply by RyanConway91** [reply]
I doubt there is a way to do this in flopy. 

best option would probably be to put the stress period data in a an external .dat and read it using open/close. Then you could easily open the stress period without flopy (in pandas) and modify it as needed.
